apps/coordinator/src/components/BCUR2/BCUR2Reader.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useRef, useMemo } from "react";
+import { BCURDecoder2, ExtendedPublicKeyData } from "@caravan/wallets";
+import { QrReader } from "react-qr-reader";
+import { Box, Button, FormHelperText, Paper, Typography } from "@mui/material";
+
+interface BCUR2ReaderProps {
+  onStart?: () => void;
+  onSuccess: (data: ExtendedPublicKeyData) => void;
+  onClear: () => void;
+  startText?: string;
+  width?: string | number;
+}
+
+const BCUR2Reader: React.FC<BCUR2ReaderProps> = ({
+  onStart,
+  onSuccess,
+  onClear,
+  startText = "Start QR Scan",
+  width = 300,
+}) => {
+  const [isScanning, setIsScanning] = useState(false);
+  const [error, setError] = useState("");
+  const decoder = useMemo(() => new BCURDecoder2(), []);
+  const statusRef = useRef<"idle" | "active" | "complete" | "error">("idle");
+
+  const handleStart = () => {
+    onStart?.();
+    setError("");
+    decoder.reset();
+    statusRef.current = "active";
+    setIsScanning(true);
+  };
+
+  const handleStop = () => {
+    decoder.reset();
+    onClear();
+    setIsScanning(false);
+    setError("");
+    statusRef.current = "idle";
+  };
+
+  const handleQRResult = (result: any, scanError: any) => {
+    if (statusRef.current !== "active") return;
+
+    if (scanError) return;
+
+    const text = result?.getText?.();
+    if (!text || !text.toLowerCase().startsWith("ur:")) return;
+
+    try {
+      console.log("Received QR part:", text);
```

Comment:
Can we clean up the console logs. At some point I think this would be good to put as visual elements in the actual component that we maybe publish as its own package. so a progress bar for parts received and then "started" and "completed" feedback.